src/index.ts

```diff
@@ -1,47 +1,23 @@
-import got from 'got'
+import axios from 'axios'
 import qs from 'qs'
 
-/**
- * The Config for the Sophos class
- * This interface allows utilization of sophos API
- * @export
- * @interface SophosConfig
- */
-interface SophosConfig {
-  clientId: string
-  clientSecret: string
-  partnerId?: string
-}
-
-// example interface for use in the sample commands
-interface Tenant {
-  id: string
-  name: string
-  dataGeography: 'US' | 'IE' | 'DE'
-  dataRegion: string
-  billingType: string
-  parter: any
-  organization: any
-  apiHost: string
-}
+import { SophosConfig, SophosTenant, SophosEndpoint } from './types'
+export { SophosConfig, SophosTenant, SophosEndpoint }
```

Comment:
I'm not sure if we need to export these types here if the TypeScript compiler is configured to export types and the `types` key in package.json points to a properly generated `.d.ts` file.

@noticeeverything could you provide some insight when you have a chance?

Reply:
We discussed this in our meeting today. No need to export here.

Reply:
@CalebAlbers @noticeeverything  so if I remove the export, the interfaces from the types.ts are only exporting in  the types.d.ts and i can't seem to access the types when using the module. And in the package.json for the types it's referencing  "./dist/index.d.ts".Here's the index.d.ts and types.d.ts

Index.d.ts
`import { SophosConfig, SophosTenant, SophosEndpoint } from './types';
export declare class Sophos {
    private readonly _config;
    domain: string;
    accessToken: string;
    constructor(_config: SophosConfig);
    getTenants(): Promise<SophosTenant[]>;
    getEndpoints(tenantId: string, tenantApiHost: string): Promise<SophosEndpoint[]>;
    private _authenticate;
    private _SophosRequest;
}
export default Sophos;
`
types.d.ts
`export interface SophosConfig {
    clientId: string;
    clientSecret: string;
    partnerId?: string;
}
export interface SophosTenant {
    id: string;
    name: string;
    dataGeography: string;
    dataRegion: string;
    billingType: string;
    partner: object;
    organization: string;
    apiHost: string;
}
export interface SophosEndpoint {
    id: string;
    type: string;
    tenant: {
        id: string;
    };
    hostname: string;
    health: {
        overall: string;
        threats: {
            status: string;
        };
        services: {
            status: string;
            serviceDetails: SophosEndpointService[];
        };
    };
    os: {
        isServer: boolean;
        platform: string;
        name: string;
        majorVersion: number;
        minorVersion: number;
        build: number;
    };
    ipv4Addresses: string[];
    macAddresses: string[];
    associatedPerson: {
        name: string;
        viaLogin: string;
        id: string;
    };
    tamperProtectionEnabled: boolean;
    assignedProducts: SophosEndpointServiceAssignedProduct[];
    lastSeenAt: string;
    encryption: {
        volumes: SophosEndpointEncryptionVolume[];
    };
}
export interface SophosEndpointService {
    name: string;
    status: string;
}
export interface SophosEndpointServiceAssignedProduct {
    code: string;
    version: string;
    status: string;
}
export interface SophosEndpointEncryptionVolume {
    volumeId: string;
    status: string;
}
export interface SophosAccessToken {
    access_token: string;
    errorCode: string;
    expires_in: string;
    message: string;
    refresh_token: string;
    token_type: string;
    trackingId: string;
}
`

Reply:
Yeah, if the idea is for consumers of this lib to be able to import everything from a barrel (e.g., `import { SophosTenant, Sophos } from 'sophos'`), the export does need to be here. These days I'd generally expect an `index.ts` file to only contain exports from various other places.